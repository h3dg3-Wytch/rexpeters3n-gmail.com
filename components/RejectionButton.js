import React from "react";
import { TouchableOpacity, Text, StyleSheet, View } from 'react-native';


function RejectionButton({ reset }) {
  return (
    <View style={styles.rejectContainer}>
            <TouchableOpacity style={styles.rejectButton}>
              <Text onPress={reset}>Restart</Text> 
            </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
    rejectButton: { 
      height: 40, 
      marginTop: 10, 
      width: 100,
      justifyContent:'center', 
      alignItems:'center', 
      backgroundColor: 'red', 
      borderRadius: 24 
    },
    rejectContainer: {
      display: 'flex', 
      justifyContent: "center", 
      alignItems: 'center'  
    }
});
export default RejectionButton;