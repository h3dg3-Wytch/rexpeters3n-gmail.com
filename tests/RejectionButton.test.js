import React from 'react';
import renderer from 'react-test-renderer';

import { shallow } from 'enzyme';

import RejectionButton from '../components/RejectionButton';
import { Text, TouchableOpacity } from 'react-native';

describe('<RejectionButton />', () => {
 
  it('can reset', () => {
    const onPressEvent = jest.fn();
      onPressEvent.mockReturnValue('Pressed');
      const component = shallow(<RejectionButton reset={onPressEvent} />);
      component.find(Text).first().props().onPress();
      expect(onPressEvent.mock.calls.length).toBe(1);
  });

});