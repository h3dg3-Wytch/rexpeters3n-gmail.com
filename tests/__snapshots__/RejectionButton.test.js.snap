// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RejectionButton /> has 1 child 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 24,
        "height": 40,
        "justifyContent": "center",
        "marginTop": 10,
        "opacity": 1,
        "width": 100,
      }
    }
  >
    <Text>
      Restart
    </Text>
  </View>
</View>
`;
